### DISCUSSION TITLE
Bump Python version to 3.10+

### DISCUSSION THREAD

**Discussion by wsw70** [jamesoff/simplemonitor#1383] [Ideas]
Python `3.10` introduced (at last!) the [`match` command](https://docs.python.org/3/reference/compound_stmts.html#match) which corresponds to the ~`switch` command in other languages.

It makes life at least 3.8 times easier.

(current version is 3.12)

**Comment by jamesoff** [comment]
Much as I like new language features (and am using them in other projects), I'm mindful of things like Raspberry Pis which may be held up on older distros with older Pythons, so I'm trying to keep compatibility with older versions (as long as they're not EOL) as much as possible :)

**Comment by wsw70** [comment]
Alllllll right - I will rewrite my code to match with 3.7 then ;-(

**Comment by wsw70** [comment]
> as long as they're not EOL 

Just FYI - 3.7 is EOL since 2023-06-27

![image](https://github.com/jamesoff/simplemonitor/assets/1345886/7f2e0ebc-a838-42f2-bcf7-192c80f45f35)

*source: https://devguide.python.org/versions/*

**Comment by jamesoff** [comment]
Time to upgrade to 3.8 then :)

**Reply by wsw70** [reply]
Yes, it buys 300 days :)

3.12, however, ... :)

**Reply by jamesoff** [reply]
Time to investigate getting my Pi3 upgraded to some newer Debian/Raspbian I think!

**Reply by wsw70** [reply]
I did that for one of my RPis (that hangs below a monitor and powers the home dashboard) and initially I went the *"change the target in `sources.list`"* way but ultimately ended up reflashing the SD card with the latest Raspberry OS (ex Raspbian). It is a Debian 12.

**Reply by jamesoff** [reply]
Yeah I just edited the sources with `s/buster/bullseye` and `dist-upgrade` is running now. Not the end of the world if I have to reflash it, it mostly just runs simplemonitor and homebridge now.
